### PR TITLE
GCS: support --send-credentials-to-tikv and get credentials from GKE or GCE

### LINKED ISSUE #106: GCS: support --send-credentials-to-tikv and get credentials from GKE or GCE
1. We can get credentials from GKE or GCE, see https://cloud.google.com/docs/authentication/production#auth-cloud-compute-engine-go as example
2. We should support --send-credentials-to-tikv when using GCS.

### PR DESCRIPTION
**Pull request by SunRunAway**
Closes #106

### CODE REVIEW

File: pkg/storage/gcs.go [hunk 1]
```diff
@@ -139,6 +139,14 @@ func (s *gcsStorage) FileExists(ctx context.Context, name string) (bool, error)
 }
 
 func newGCSStorage(ctx context.Context, gcs *backup.GCS) (*gcsStorage, error) {
+	if gcs.CredentialsBlob == "" {
```
Comment (overvenus) [inline pkg/storage/gcs.go]:
```suggestion
	if gcs.CredentialsBlob == ""  && sendCredential {
```

**Comment by gregwebs** [comment]
LGTM

File: pkg/storage/gcs.go (hunk 1 above)
Comment (SunRunAway) [reply pkg/storage/gcs.go]: No, it's not related to `sendCredential`.
We should get credentials from environment if user does not provide credentials explicitly.

**Comment by SunRunAway** [comment]
> Could you add some unit tests to improve coverage?

Ok, I'll try.

**Comment by overvenus** [comment]
Could you fix conflicts?

**Pull request merged**